agent/lib/broker_controller.js

```diff
@@ -426,14 +488,17 @@ BrokerController.prototype.delete_address_and_settings = function (a) {
         log.info('[%s] Deleted address-settings for "%s"', self.id, a.address);
         return self.delete_address(a);
     }).catch(function (error) {
-        log.error('[%s] Failed to delete address settings for "%s": %s', self.id, a.address, error);
+        log.error('[%s] Failed to delete address setting for "%s": %s', self.id, a.address, error);
```

Comment:
I don't understand the need for the two catches.  Would moving the self.delete_address(a) to a .finally() make the code simpler?

Reply:
removed the extra